File: kaleido-gradle-plugin/src/main/java/com/tongsr/kaleido/gradle/dsl/KaleidoSigning.java
package com.tongsr.kaleido.gradle.dsl;

import org.gradle.api.file.RegularFileProperty;
import org.gradle.api.provider.Property;

public abstract class KaleidoSigning {
    public abstract RegularFileProperty getKeyStoreFile();

    public abstract Property<String> getStorePassword();

    public abstract Property<String> getKeyAlias();

    public abstract Property<String> getKeyPassword();

    public abstract Property<String> getExpectedCertificateSha256();
}

File: kaleido-gradle-plugin/src/main/kotlin/com/tongsr/kaleido/gradle/dsl/KaleidoSigning.kt
package com.tongsr.kaleido.gradle.dsl

import org.gradle.api.file.RegularFileProperty
import org.gradle.api.provider.Property

abstract class KaleidoSigning {
    abstract val keyStoreFile: RegularFileProperty
    abstract val storePassword: Property<String>
    abstract val keyAlias: Property<String>
    abstract val keyPassword: Property<String>
    abstract val expectedCertificateSha256: Property<String>
}
